### DISCUSSION TITLE
Is there any way to get all the replies of a message?

### DISCUSSION THREAD

**Discussion by zilewang7** [telegraf/telegraf#2013] [Q&A] [answered]
Just like the View Replies button in the message menu in the client.

Is there any simple implementation, or do I need to maintain the relationships between messages in the database myself?

**Comment by MKRhere** [comment, marked as answer]
>do I need to maintain the relationships between messages in the database myself?

Yes. The Bot API does not provide any way to retrieve messages except by events.
